dbuf-lsp/src/handler.rs

```diff
@@ -4,9 +4,14 @@
 use tower_lsp::lsp_types::InitializeParams;
 use tower_lsp::lsp_types::ServerCapabilities;
 
+/// Capabilities of handler
+pub trait Capabilities {
+    fn apply(self, capabilities: &mut ServerCapabilities);
```

Comment:
Well, the reason I commented on your approach to initialization of capabilities is _because_ you pass a `&mut` which can change across several initialization functions and these changes may accidentally turn out to be inconsistent or messing up each other's results. So the solution I proposed was to build up some custom small type of the capabilities you support (which is built up from parts from different parts of an LSP) and the `ServerCapabilities` object is built up from that, so that the possible conflicts are resolved in a single place, where the conversion from your type to `ServerCapabilities` happens

Reply:
The best approach, I guess, would be to simply return parts of `ServerCapabilities` from `.init` functions -- for example, an `ActionHandler::init` would return a `(OneOf<bool, DocumentFormattingOptions>, RenameOptions)`